### PR TITLE
Immutable field

### PR DESCRIPTION
**Pull request by ak708**
closes #

## 📑 Description
Backend - Immutable added to Field type, and property in ctrl
Frontend - All edit module inputs and switches can now be disabled using immutable in json
## ✅ Checks
- [ ] I have updated the documentation as required
- [ ] I have performed a self-review of my code

### CODE REVIEW

File: cyclops-ctrl/docker-compose.yaml [hunk 1]
```diff
@@ -4,4 +4,4 @@ services:
     container_name: redis
     image: redis
     ports:
-      - 6379:6379
\ No newline at end of file
+      - 8888:8888
```
Comment (petar-cvit) [inline cyclops-ctrl/docker-compose.yaml:7]: Can you revert this?

File: cyclops-ctrl/internal/models/templates.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ type Field struct {
 	Enum          []interface{} `json:"enum"`
 	Required      []string      `json:"required"`
 	FileExtension string        `json:"fileExtension"`
+	Immutable	  bool          `json:"immutable"`
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/models/templates.go:42]: Can you run 'go fmt ./...' on cyclops-ctrl?

File: cyclops-ui/pnpm-lock.yaml
Comment (petar-cvit) [inline cyclops-ui/pnpm-lock.yaml:1]: You can remove this file

**Review by petar-cvit — CHANGES_REQUESTED** [review]
Thanks @ak708, check out the comments below

File: cyclops-ctrl/docker-compose.yaml (hunk 1 above)
Comment (ak708) [reply cyclops-ctrl/docker-compose.yaml:7]: i changed it to       - 6379:6379

File: cyclops-ctrl/internal/models/templates.go (hunk 2 above)
Comment (ak708) [reply cyclops-ctrl/internal/models/templates.go:42]: ran it, it downloaded a bunch of modules

File: cyclops-ui/pnpm-lock.yaml
Comment (ak708) [reply cyclops-ui/pnpm-lock.yaml:1]: done, do i push and raise a pr now?

Comment (petar-cvit) [reply cyclops-ui/pnpm-lock.yaml:1]: You can just push to this branch

**Comment by petar-cvit** [comment]
@ak708 can you also commit other changes?

**Comment by ak708** [comment]
@petar-cvit others? you mean for other issues?

**Comment by petar-cvit** [comment]
I meant commit changes for deleting `[cyclops-ui/pnpm-lock.yaml](https://github.com/cyclops-ui/cyclops/pull/494/files/81ac7789776713eb98c682a07dd2a5bfc3e09dd7#diff-2a8515163b5172d1c13b6c14ed0e36eb66016af49ad608759527b40a1e35ed12)` and running [go fmt](https://github.com/cyclops-ui/cyclops/pull/494#discussion_r1703149514)

**Comment by ak708** [comment]
the latest commit does not have the pnpm-lock and go fmt has finished running. it shows that maybe due to previous commits, should i close this and open a new pull request?

File: cyclops-ctrl/internal/mapper/helm.go [hunk 3]
```diff
@@ -88,6 +90,7 @@ func HelmSchemaToFields(name string, schema helm.Property, defs map[string]helm.
 		MinLength:        schema.MinLength,
 		MaxLength:        schema.MaxLength,
 		Pattern:          schema.Pattern,
+		Immutable         schema.Immutable,
```
Comment (petar-cvit) [inline cyclops-ctrl/internal/mapper/helm.go:93]: Missing `:`

**Review by petar-cvit — CHANGES_REQUESTED** [review]

File: cyclops-ctrl/internal/mapper/helm.go (hunk 3 above)
Comment (ak708) [reply cyclops-ctrl/internal/mapper/helm.go:93]: @petar-cvit sorry, added it

**Comment by petar-cvit** [comment]
Is there a reason to close this one? You can keep this one open. If you need to rename it feel free to do it, but Im fine with this one

**Review by petar-cvit — APPROVED** [review]
Looks good @ak708! Thanks 🧡

**Pull request merged**